eth/db/journal.py

```diff
@@ -384,6 +384,8 @@ def commit(self, changeset_id: uuid.UUID) -> None:
 
         If this is the base changeset then all changes will be written to
         the underlying database and the Journal starts a new recording.
+        Typically, callers won't have access to the base changeset, because
+        it is dropped during .reset() which is called in JournalDB().
```

Comment:
Nice addition :) I thought I had found a bug until I saw this comment, we probably went down the same path :laughing:

Reply:
I'm starting to think that `commit()` does the wrong thing. `commit()` should only ever collapse changesets together, and the entire `self.journal.is_empty()` block should move into `persist()`. That would be more straightforward.

Reply:
> Nice addition :) I thought I had found a bug until I saw this comment, we probably went down the same path

Haha, yeah, I remember that surprising me a bit.

> I'm starting to think that `commit()` does the wrong thing. `commit()` should only ever collapse changesets together, and the entire `self.journal.is_empty()` block should move into `persist()`. That would be more straightforward.

I think I'm on board, but probably belongs in a different PR. Opening an issue.